### ISSUE TITLE
Solution for heatmap shrinkage - examples improvement

### ISSUE THREAD

**Issue by srjnm** [markus-wa/demoinfocs-golang#303]
**Describe the bug**
The heatmap generated by [heatmap.go](https://github.com/markus-wa/demoinfocs-golang/blob/master/examples/heatmap/heatmap.go) gets shrinked if the size of dots are increased and goes out of the map bounds. This might be happening due to the bounding rectangle being too pixel perfect, which causes [go-heatmap](https://github.com/dustin/go-heatmap) to scale down the points.

**Solution**
I expanded the bounding rectangle by 16 units.
I had to come up with a brute-force solution out of frustration, but it fixed my problem quite well. The value 16 units most likely wouldn't fix your case, but hopefully it'll help you figure out your own value or come up with a better solution.

Original Dot Size: 15
My Dot Size: 30

Code:
lines 63-68: [heatmap.go](https://github.com/markus-wa/demoinfocs-golang/blob/master/examples/heatmap/heatmap.go)
Current-
```
       // Find bounding rectangle for points to get around the normalization done by the heatmap library
       r2Bounds := r2.RectFromPoints(points...)
       bounds := image.Rectangle{
       		Min: image.Point{X: int(r2Bounds.X.Lo), Y: int(r2Bounds.Y.Lo)},
       		Max: image.Point{X: int(r2Bounds.X.Hi), Y: int(r2Bounds.Y.Hi)},
       }
```

Changes-
```
       // Find bounding rectangle for points to get around the normalization done by the heatmap library
       r2Bounds := r2.RectFromPoints(points...)
       bounds := image.Rectangle{
       		Min: image.Point{X: int(r2Bounds.X.Lo - 16), Y: int(r2Bounds.Y.Lo - 16)},
       		Max: image.Point{X: int(r2Bounds.X.Hi + 16), Y: int(r2Bounds.Y.Hi + 16)},
       }
```

**Results**
Before - lines look messed up
![before](https://user-images.githubusercontent.com/54297985/122601332-c12fc980-d08e-11eb-83b2-fac753776d24.jpeg)
After - lines were perfect (:
![after](https://user-images.githubusercontent.com/54297985/122601358-cbea5e80-d08e-11eb-89aa-df6aa78f9b5e.jpeg)

**Comment by markus-wa** [comment]
Interesting, well spotted @srjnm - what about `dotSize/2` instead of 16, considering your dot-size is 30 and 15/16px is basically the same? or is that much worse?

Also, would you be able to create a PR with this fix?

**Comment by srjnm** [comment]
> Interesting, well spotted @srjnm - what about `dotSize/2` instead of 16, considering your dot-size is 30 and 15/16px is basically the same? or is that much worse?

argh should have noticed that, my bad. I checked it with different sizes and it works.

> Also, would you be able to create a PR with this fix?

yes definitely, I'll ping the PR here after I'm done.

**Comment by srjnm** [comment]
`os.Args = []string{"cmd", "-demo", "../../test/cs-demos/default.dem"}`

@markus-wa Can you just pass this demo file so that I can redo the output image?

**Comment by srjnm** [comment]
nvm, I got it [here](https://gitlab.com/markus-wa/cs-demos-2/-/blob/master/default.7z)

**Comment by markus-wa** [comment]
Glad you found it, btw. you can use the scripts in the `scripts` folder to download demos or check out the readme section about development. The reason they aren't included by default is because they are large and go is a bit weird 😅

**Comment by markus-wa** [comment]
Closed by #304 🚀
